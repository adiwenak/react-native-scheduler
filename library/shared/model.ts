export enum Month {
  January = 0,
  February,
  March,
  April,
  May,
  June,
  July,
  August,
  September,
  October,
  November,
  December
}

export type TimeInMinutes = number
export type CategoryId = string
export type ActivityId = string

// tslint:disable-next-line:max-line-length
export type DateNumber = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 | 25 | 26 | 27 | 28 | 29 | 30 | 31

export interface BaseTimeline {
  startTime: Date
  endTime: Date
}

export interface Category {
  id: CategoryId
  name: string
  colour: string
}

export interface CategorySelection extends Category {
  isSelected: boolean
}

export interface Activity extends BaseTimeline {
  id: ActivityId
  category: CategoryId
  name: string
}

// CategoryId map to activities
export interface CategoryActivities {
  [key: string]: Activity[]
}

export interface CategoryActivity {
  category: CategorySelection
  tasks: Activity[]
}
