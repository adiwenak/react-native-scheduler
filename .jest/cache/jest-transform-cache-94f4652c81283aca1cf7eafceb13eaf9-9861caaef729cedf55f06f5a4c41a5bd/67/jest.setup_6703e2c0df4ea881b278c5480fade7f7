eabf2f65e7ae4e05e4ba679363e1ad5a
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");var _enzymeAdapterReact = _interopRequireDefault(require("enzyme-adapter-react-16"));
var _enzyme = _interopRequireDefault(require("enzyme"));

_enzyme.default.configure({ adapter: new _enzymeAdapterReact.default() });