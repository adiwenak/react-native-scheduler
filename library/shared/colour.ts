// define the name and pick it up from here
// http://chir.ag/projects/name-that-color

export const Colours = {
    // Blue
    Cobalt: "#054BBD",

    // Green
    LaPalma: "#149427",

    // Orange
    Zest: "#E4741B",

    // Red
    Cardinal: "#CC0F20"
}
