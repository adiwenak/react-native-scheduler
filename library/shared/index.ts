export * from "./model"
export * from "./colour"
