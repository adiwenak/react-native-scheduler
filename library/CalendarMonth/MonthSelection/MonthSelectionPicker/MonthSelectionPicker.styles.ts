import { StyleSheet, TextStyle, ViewStyle } from "react-native"

import { darkGrey, midGrey } from "../../../shared/colour"

interface Style {
  monthTitle: ViewStyle
  monthTitleText: TextStyle
  containerBox: ViewStyle
  individualContainer: ViewStyle
  monthFont: TextStyle
  outerView: ViewStyle
}

export const styles = StyleSheet.create<Style>({
    monthTitle: {
        height: 40,
        alignSelf: "center",
        justifyContent: "center"
    },
    monthTitleText: {
        fontSize: 16,
        color: darkGrey
    },
    containerBox: {
        flexDirection: "row",
        flexWrap: "wrap"
    },
    individualContainer: {
        margin: 2,
        backgroundColor: "white",
        height: 60,
        width: 45,
        borderRadius: 5,
        justifyContent: "center"
    },
    monthFont: {
        fontSize: 14,
        textAlign: "center",
        color: darkGrey
    },
    outerView: {
        flexDirection: "column",
        width: 200,
        height: 240,
        backgroundColor: "white",
        borderRadius: 5,
        borderWidth: 2,
        borderColor: "lightgrey"
    }
})
