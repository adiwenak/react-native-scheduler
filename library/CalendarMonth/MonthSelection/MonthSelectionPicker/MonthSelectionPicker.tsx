import moment from "moment"
import * as React from "react"
import { FlatList, Text, TouchableOpacity, View } from "react-native"

import { oceanBlue } from "../../../shared/colour"
import { Month } from "../../../shared/model"
import { styles } from "./MonthSelectionPicker.styles"

interface MonthElement {
    month: string
    monthNum: Month
}

interface ComponentState {
    showYearView: boolean
}

interface ComponentProps {
    onMonthChangeFromPicker: (month: Month) => void
    onYearChangeFromPicker: (year: number) => void
    currentMonth: Month
    currentYear: number
    startYear: number
    endYear: number
}

export class MonthSelectionPicker extends React.Component<ComponentProps, ComponentState> {

    private monthElements: MonthElement[] = [
        { month: "Jan", monthNum: Month.January },
        { month: "Feb", monthNum: Month.February },
        { month: "Mar", monthNum: Month.March },
        { month: "Apr", monthNum: Month.April },
        { month: "May", monthNum: Month.May },
        { month: "Jun", monthNum: Month.June },
        { month: "Jul", monthNum: Month.July },
        { month: "Aug", monthNum: Month.August },
        { month: "Sep", monthNum: Month.September },
        { month: "Oct", monthNum: Month.October },
        { month: "Nov", monthNum: Month.November },
        { month: "Dec", monthNum: Month.December }
    ]

    private yearList: number[] = []

    constructor(props: ComponentProps) {
        super(props)
        this.state = {
            showYearView: false
        }

        for (let i = this.props.endYear; i >= this.props.startYear; i--) {
            this.yearList.push(i)
        }
    }

    public renderYearList = () => {
        return (
            <FlatList
                data={this.yearList}
                keyExtractor={(_, index) => `${index}`}
                renderItem={this.renderYearListItem}
                numColumns={4}
                extraData={this.props.currentYear}
            />
        )
    }

    public renderYearListItem = ({item}: any) => (
        <TouchableOpacity
            accessibilityLabel={item.toString()}
            style={[styles.individualContainer,
                this.props.currentYear === item ?
                { backgroundColor: oceanBlue } : null ]}
            onPress={() => {
                this.props.onYearChangeFromPicker(item)
            }}
        >
            <Text
                style={[styles.monthFont, this.props.currentYear === item ? { color: "white" } : null]}
            >
                {item}
            </Text>
        </TouchableOpacity>
    )

    public renderMonthList = () => {
        return (
            <View style={styles.containerBox}>
                {this.monthElements.map((monthObj: MonthElement) => (
                    <TouchableOpacity
                        key={monthObj.month}
                        accessibilityLabel={monthObj.month}
                        style={[styles.individualContainer,
                            this.props.currentMonth === monthObj.monthNum ?
                            { backgroundColor: oceanBlue } : null ]}
                        onPress={() => {
                            this.props.onMonthChangeFromPicker(monthObj.monthNum)
                        }}
                    >
                        <Text style={[styles.monthFont, this.props.currentMonth === monthObj.monthNum ?
                        { color: "white" } : null]}>
                            {monthObj.month}
                        </Text>
                    </TouchableOpacity>
                ))}
            </View>
        )
    }

    public render(): JSX.Element {
        return (
            <View style={styles.outerView}>
                <TouchableOpacity
                    style={styles.monthTitle}
                    accessibilityLabel={"month-year-btn"}
                    onPress={() => this.setState({ showYearView: !this.state.showYearView })}
                >
                    <Text style={styles.monthTitleText}>{this.state.showYearView ? "Year" : "Month"}</Text>
                </TouchableOpacity>
                {this.state.showYearView ? this.renderYearList() : this.renderMonthList()}
            </View>
        )
    }
}
