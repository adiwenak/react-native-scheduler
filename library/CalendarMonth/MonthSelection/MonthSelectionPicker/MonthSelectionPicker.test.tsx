import { shallow } from "enzyme"
import { default as React } from "react"

import "jest"
import "react-native"

import { Month } from "../../../shared/model"
import { MonthSelectionPicker } from "./MonthSelectionPicker"

describe("MonthSelectionPicker.tsx", () => {
  describe("<MonthSelectionPicker />", () => {
    let onMonthChangeFromPicker: any
    let onYearChangeFromPicker: any
    let comp: any

    beforeEach(() => {
      onMonthChangeFromPicker = jest.fn()
      onYearChangeFromPicker = jest.fn()

      comp = shallow(
        <MonthSelectionPicker
          onMonthChangeFromPicker={onMonthChangeFromPicker}
          onYearChangeFromPicker={onYearChangeFromPicker}
          currentMonth={Month.January}
          currentYear={2019}
          startYear={2010}
          endYear={2021}
          />)
    })
    it("should render with proper props", () => {
      expect(comp).toMatchSnapshot()
    })
    it("should call onMonthChangeFromPicker when pressed - Feb", () => {
      const touchable = comp.find({ accessibilityLabel: "Feb" }).props()

      touchable.onPress()
      expect(onMonthChangeFromPicker).toBeCalledTimes(1)
      expect(onMonthChangeFromPicker).toBeCalledWith(Month.February)
    })
    it("should call onYearChangeFromPicker when pressed - 2018", () => {
      const touchable = comp.find({ accessibilityLabel: "month-year-btn" }).props()
      expect(comp.instance().state).toEqual({ showYearView: false })
      touchable.onPress()
      expect(comp.instance().state).toEqual({ showYearView: true })
    })
    it("should call onYearChnageFromPicker when renderYearListItem is pressed", () => {
      const button = comp.instance().renderYearListItem({ item: 2019 }).props.onPress
      button()
      expect(onYearChangeFromPicker).toBeCalledTimes(1)
      expect(onYearChangeFromPicker).toBeCalledWith(2019)
    })
    it("keyExtractor", () => {
      const keyExtractor = comp.instance().renderYearList().props.keyExtractor
      expect(keyExtractor("1", 1)).toEqual("1")
    })
  })
})
