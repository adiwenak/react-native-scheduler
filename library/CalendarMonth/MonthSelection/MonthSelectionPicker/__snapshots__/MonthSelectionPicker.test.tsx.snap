// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthSelectionPicker.tsx <MonthSelectionPicker /> should render with proper props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "lightgrey",
      "borderRadius": 5,
      "borderWidth": 2,
      "flexDirection": "column",
      "height": 240,
      "width": 200,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "height": 40,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(100, 100, 100)",
          "fontSize": 16,
        }
      }
    >
      Month
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityLabel="Jan"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(26, 117, 173)",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Jan
      </Text>
    </View>
    <View
      accessibilityLabel="Feb"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Feb
      </Text>
    </View>
    <View
      accessibilityLabel="Mar"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Mar
      </Text>
    </View>
    <View
      accessibilityLabel="Apr"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Apr
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityLabel="May"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        May
      </Text>
    </View>
    <View
      accessibilityLabel="Jun"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Jun
      </Text>
    </View>
    <View
      accessibilityLabel="Jul"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Jul
      </Text>
    </View>
    <View
      accessibilityLabel="Aug"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Aug
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityLabel="Sep"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Sep
      </Text>
    </View>
    <View
      accessibilityLabel="Oct"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Oct
      </Text>
    </View>
    <View
      accessibilityLabel="Nov"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Nov
      </Text>
    </View>
    <View
      accessibilityLabel="Dec"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "margin": "1.5%",
          "opacity": 1,
          "width": "22%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(100, 100, 100)",
              "fontSize": 14,
              "textAlign": "center",
            },
            null,
          ]
        }
      >
        Dec
      </Text>
    </View>
  </View>
</View>
`;
